docs/csharp/programming-guide/events/how-to-publish-events-that-conform-to-net-framework-guidelines.md

```diff
@@ -1,75 +1,77 @@
+
+The name `EventHandler` can lead to a bit of confusion, as the delegate itself doesn't actually handle the event but instead is used to raise the event. The <xref:System.EventHandler>, and generic <xref:System.EventHandler%601> are intended to represent a method that will handle an event.
+
+### To publish events based on the EventHandler pattern
+
+1. (Skip this step and go to Step 3a if you do not have to send custom data with your event.) Declare the class for your custom data at a scope that is visible to both your publisher and subscriber classes. Then add the required members to hold your custom event data. In this example, a simple string is returned.
+
+    ```csharp
+    public class CustomEventArgs : EventArgs
+    {
+        public CustomEventArgs(string message)
+        {
+            Message = message;
         }
-    }  
-    ```  
-  
-2. (Skip this step if you are using the generic version of <xref:System.EventHandler%601> .) Declare a delegate in your publishing class. Give it a name that ends with *EventHandler*. The second parameter specifies your custom EventArgs type.  
-  
-    ```csharp  
-    public delegate void CustomEventHandler(object sender, CustomEventArgs a);  
-    ```  
-  
-3. Declare the event in your publishing class by using one of the following steps.  
-  
-    1. If you have no custom EventArgs class, your Event type will be the non-generic EventHandler delegate. You do not have to declare the delegate because it is already declared in the <xref:System> namespace that is included when you create your C# project. Add the following code to your publisher class.  
-  
-        ```csharp  
-        public event EventHandler RaiseCustomEvent;  
-        ```  
-  
-    2. If you are using the non-generic version of <xref:System.EventHandler> and you have a custom class derived from <xref:System.EventArgs>, declare your event inside your publishing class and use your delegate from step 2 as the type.  
-  
-        ```csharp  
-        public event CustomEventHandler RaiseCustomEvent;  
-        ```  
-  
-    3. If you are using the generic version, you do not need a custom delegate. Instead, in your publishing class, you specify your event type as `EventHandler<CustomEventArgs>`, substituting the name of your own class between the angle brackets.  
-  
-        ```csharp  
-        public event EventHandler<CustomEventArgs> RaiseCustomEvent;  
-        ```  
-  
-## Example  
- The following example demonstrates the previous steps by using a custom EventArgs class and <xref:System.EventHandler%601> as the event type.  
-  
- [!code-csharp[csProgGuideEvents#2](~/samples/snippets/csharp/VS_Snippets_VBCSharp/csProgGuideEvents/CS/Events.cs#2)]  
-  
+
+        public string Message { get; set; }
```

Comment:
I'd make this readonly:

```suggestion
        public string Message { get; }
```

Reply:
I did initially, and then it broke some code. It is appended to later, see [line 222](https://github.com/dotnet/docs/pull/18642/files#diff-58e13354452c4c47d554013a02f59919R222)